gdal/frmts/raw/byndataset.cpp

```diff
@@ -0,0 +1,766 @@
+/*                              SetScale()                              */
+/************************************************************************/
+
+CPLErr BYNRasterBand::SetScale( double dfNewValue )
+{
+    BYNDataset *poIDS = reinterpret_cast<BYNDataset *>( poDS );
+
+    dfNewValue = 1.0 / dfNewValue;
+
+    if( dfNewValue == poIDS->hHeader.dfFactor )
+        return CE_None;
+
+    poIDS->hHeader.dfFactor = dfNewValue;
+    return CE_None;
+}
+
+/************************************************************************/
+/* ==================================================================== */
+/*                              BYNDataset                              */
+/* ==================================================================== */
+/************************************************************************/
+
+BYNDataset::BYNDataset() : 
+        fpImage(nullptr),
+        hHeader{0,0,0,0,0,0,0,0,0.0,0,0,0,0,0,0,0,0,0.0,0.0,0,0,0.0,0}
+{
+    adfGeoTransform[0] = 0.0;
+    adfGeoTransform[1] = 1.0;
+    adfGeoTransform[2] = 0.0;
+    adfGeoTransform[3] = 0.0;
+    adfGeoTransform[4] = 0.0;
+    adfGeoTransform[5] = 1.0;
+}
+
+/************************************************************************/
+/*                            ~BYNDataset()                             */
+/************************************************************************/
+
+BYNDataset::~BYNDataset()
+
+{
+    FlushCache();
+
+    if( GetAccess() == GA_Update)
+        UpdateHeader();
+
+    SetMetadataItem("GLOBAL",     CPLSPrintf("%d",hHeader.nGlobal), "BYN");
+    SetMetadataItem("TYPE",       CPLSPrintf("%d",hHeader.nType),   "BYN");
+    SetMetadataItem("DESCRIPTION",CPLSPrintf("%d",hHeader.nDescrip),"BYN");
+    SetMetadataItem("SUBTYPE",    CPLSPrintf("%d",hHeader.nSubType),"BYN");
+    SetMetadataItem("WO",         CPLSPrintf("%g",hHeader.dfWo),    "BYN");
+    SetMetadataItem("GM",         CPLSPrintf("%g",hHeader.dfGM),    "BYN");
+    SetMetadataItem("TIDESYSTEM", CPLSPrintf("%d",hHeader.nTideSys),"BYN");
+    SetMetadataItem("REALIZATION",CPLSPrintf("%d",hHeader.nRealiz), "BYN");
+    SetMetadataItem("EPOCH",      CPLSPrintf("%g",hHeader.dEpoch),  "BYN");
+    SetMetadataItem("PTTYPE",     CPLSPrintf("%d",hHeader.nPtType), "BYN");
+
+    if( fpImage != nullptr )
+    {
+        if( VSIFCloseL( fpImage ) != 0 )
+        {
+            CPLError( CE_Failure, CPLE_FileIO, "I/O error" );
+        }
+    }
+}
+
+/************************************************************************/
+/*                              Identify()                              */
+/************************************************************************/
+
+int BYNDataset::Identify( GDALOpenInfo *poOpenInfo )
+
+{
+    if( poOpenInfo->nHeaderBytes < BYN_HDR_SZ )
+        return FALSE;
+
+    if( ! ( EQUAL(CPLGetExtension(poOpenInfo->pszFilename), "byn") ||
+            EQUAL(CPLGetExtension(poOpenInfo->pszFilename), "err") ))
+        return FALSE;
+
```

Comment:
As matching by extension can be fragile, can you check the values of a few header fields as well ?

Reply:
The header doesn't have any particular identification. The only thing I can think about is to check if the South, North, West, East fields are in a reasonable value range and/or the, therefore, the computation of the XSize and YSize result in reasonable positive values. Is that OK? Would that be fast enough for the driver's manager probe?

Reply:
Yes, and I also see there are a number of fields with enumerated values that would be good candidates